source/developer/interactive-message-buttons.rst

```diff
@@ -0,0 +1,116 @@
+            "integration": {
+              "url": "http://127.0.0.1:7357",
+              "context": {
+                "action": "ephemeral"
+              }
+            }
+          }, {
+            "name": "Update",
+            "integration": {
+              "url": "http://127.0.0.1:7357",
+              "context": {
+                "action": "update"
+              }
+            }
+          }
+        ]
+      }
+    ]
+  }
+
+The integration can respond with an update to the original post, or with an ephemeral message:
+
+.. code-block:: text
+
+  {
+    "update": {
+      "message": "Updated!"
+    },
+    "ephemeral_text": "You updated the post!"
+  }
+
+// XXX Insert an image here
+
+Below we give a brief description of each parameter to help you customize the webhook post in Mattermost. For more information on message attachments, `see our documentation <https://docs.mattermost.com/developer/message-attachments.html>`_.
+
+Name
+  Give your action a descriptive name.
+
+URL
+  The actions are backed by an integration that handles HTTP POST requests when users clicks the message button. The URL parameter determines where this action is sent to. The request contains an ``application/x-www-form-urlencoded`` JSON string. // XXX @ccbrown is this correct?
+
+Context // XXX @ccbrown Need a bit of help to more specifically explain this parameter.
+  The requests sent to the specified URL contain the user id and any context that was provided in the action definition as follows:
+  
+  .. code-block:: text
+
+    {
+    "user_id": "rd49ehbqyjytddasoownkuqrxe",
+    "context": {
+      "action": "ephemeral"
+      }
+    }
+
+Tips and Best Practices
+------------------------
+
+// XXX @ccbrown Do we want to clean up the polling integration demo and provide it as an example? https://github.com/ccbrown/interactive-post-demo
+
+1. The external application may be written in any programming language. It needs to provide a URL which receives the request sent by your Mattermost server and responds with in the required JSON format.
+2. Message attachments allow for rich formatting, but don't go overboard. Use the least amount of formatting and number of action buttons required for your integration post.
+
+// XXX @ccbrown
+
+Share Your Integration
+-----------------------
+
+If you've built an integration for Mattermost, please consider `sharing your work <https://www.mattermost.org/share-your-mattermost-projects/>`_ in our `app directory <https://about.mattermost.com/default-app-directory/>`_.
+
+The `app directory <https://about.mattermost.com/default-app-directory/>`_ lists open source integrations developed by the Mattermost community and are available for download, customization and deployment to your private cloud or on-prem infrastructure.
+
+Slack Compatibility
+--------------------
+
+// XXX @ccbrown, please help with how one can turn a Slack-type message button into a Mattermost one? It was just a few tweaks if I remember correctly?
```

Comment:
Hmm. They might be a little too distant from each other for a concise guide... The format for the message attachment JSON is somewhat close, but the integration request and response are totally different.

Reply:
Maybe something like...

> Interactive messages are not intended to be Slack compatible.
> 
> Slack requires a Slack App and action URL to be pre-configured beforehand. Mattermost allows any webhook or slash command to create an interactive message without any additional configuration.
> 
> Like Slack, actions are specified in an "actions" list within the Slack attachment. However, the schema for these objects is completely different.
> 
> With Slack, the request made to your integration when a user performs an action contains information such as channel and team ids. With Mattermost, it contains only a user id and whatever information you specified in your context. You can put the team id or channel id in your context when you post the message if you'll need these things again later.
> 
> Like Slack, your integrations can react with ephemeral messages or message updates. But again, the schema for this is slightly different.

:-/